### ISSUE TITLE
Better tests for Waku Protocol

### ISSUE THREAD

**Issue by kdeme** [status-im/nim-eth#169]
- Waku should take over the tests from test_shh_connect.nim which are applicable.
- Some rework/clean-up should be done to always use a future to wait for when posting messages where possible (this approach is taken often but not always yet)
- The Waku handshake should be more extensively tested with different options omitted, or currently not existing options added. 

It would be wise though to wait with these changes until https://github.com/status-im/nim-eth/issues/168 us done, as this might introduce several changes. Also, that new packet type itself needs to be tested.

**Comment by zah** [comment]
It should be possible to make the tests generic and specify the Protocol/Capability type as a parameter. We'll be then able to run them with both protocols.

**Comment by kdeme** [comment]
Yes, that should be possible. Not all tests in test_shh_connect.nim are valid tests for Waku though so some need to be split.
